r/tools/nixlibs.R

```diff
@@ -814,8 +822,16 @@ set_thirdparty_urls <- function(env_var_list) {
   env_var_list
 }
 
-is_feature_requested <- function(env_varname, default = env_is("LIBARROW_MINIMAL", "false")) {
-  env_value <- tolower(Sys.getenv(env_varname))
+# this is generally about features that people asked for via environment variables, but
+# for some cases (like S3 when we override it in this script) we might find those in
+# env_var_list
+is_feature_requested <- function(env_varname, env_var_list, default = env_is("LIBARROW_MINIMAL", "false")) {
+  # look in the environment first, but then use the env_var_list if nothing is found
+  env_var_list_value <- env_var_list[[env_varname]]
+  if (is.null(env_var_list_value)) {
+    env_var_list_value <- ""
+  }
+  env_value <- tolower(Sys.getenv(env_varname, env_var_list_value))
```

Comment:
Is this equivalent?

```suggestion
  env_value <- tolower(Sys.getenv(env_varname, env_var_list[[env_varname]]))
```

Reply:
Well I feel slightly better having thought the exact same thing. But turns out: no! `env_var_list[[env_varname]]` is `NULL` if it doesn't exist (which is what we want) but:

```
Sys.getenv("env_varname", NULL)
Error in Sys.getenv("env_varname", NULL) : wrong type for argument
```

Though I only found that out when I pushed (+ got a little too eager and triggered all of crossbow r) and then saw lots of red.

This would be a perfect place for `%||%` or the native equivalent, but I didn't want to require rlang here (I mean we require it for the package, so maybe that's ok?)